engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -207,6 +215,13 @@ protected boolean filter(ExcludeList avoid, StoragePool pool, DiskProfile dskCh,
             return false;
         }
 
+        if (diskProvisioningTypeStrictness){
+            StoragePoolDetailVO hardwareAcceleration = storagePoolDetailsDao.findDetail(pool.getId(), "hardwareAccelerationSupported");
```

Comment:
it is better to define a string constant for "`hardwareAccelerationSupported`", in some common storage / util class and use it across.

Reply:
Done